### PR TITLE
Reskin card type chooser according to designs

### PR DESCRIPTION
**Pull request by burieberry**
[CS-5445](https://linear.app/cardstack/issue/CS-5445/re-skin-card-type-chooser)

- Can select card
- Buttons display on hover
- Added footer with cardURL input, cancel and go buttons
- Update tests to use the go button after card selection

card-type-catalog

### CODE REVIEW

File: packages/host/app/components/card-catalog-modal.gts [hunk 1]
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
                   data-test-create-new
                 >Create New</Button>
               {{/if}}
+              <div class='card-catalog__results-length'>{{if
+                  this.currentRequest.search.instances.length
+                  this.currentRequest.search.instances.length
+                  '0'
+                }}
+                results</div>
```
Comment (jurgenwerk) [inline packages/host/app/components/card-catalog-modal.gts]: There can also be only 1 result and we should show the singular form of "results" in that case

**Review by jurgenwerk — APPROVED** [review]
Minor pluralization thing but otherwise looks good

File: packages/host/app/components/card-catalog-modal.gts (hunk 1 above)
Comment (jurgenwerk) [reply packages/host/app/components/card-catalog-modal.gts]: Also I think "results" shouldn't be shown without a number in front of it, so either "x results" or nothing. If I understand correctly currently with this code there will be a moment where only "results" will be shown when there is no number yet?

File: packages/host/app/components/card-catalog-modal.gts [hunk 2]
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
...
               <ul class='card-catalog' data-test-card-catalog>
                 {{#each this.currentRequest.search.instances as |card|}}
-                  <li data-test-card-catalog-item={{card.id}}>
+                  <li
+                    class={{cn
+                      'card-catalog-item'
+                      is-selected=(eq this.selectedCard.id card.id)
```
Comment (jurgenwerk) [inline packages/host/app/components/card-catalog-modal.gts]: Maybe modifier css class would be more appropriate? `card-catalog-item--selected`

File: packages/host/app/components/card-catalog-modal.gts [hunk 3]
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
...
+                  >
                     <Preview
                       @card={{card}}
                       @format='embedded'
                       @context={{@context}}
                     />
                     <button
-                      {{on 'click' (fn this.pick card)}}
+                      class='card-catalog-item__select'
+                      {{on 'click' (fn this.select card)}}
```
Comment (lukemelia) [inline packages/host/app/components/card-catalog-modal.gts]: Can you unselect a card once it is selected?

File: packages/host/app/components/card-catalog-modal.gts [hunk 4]
```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
...
+          <footer class='dialog-box__footer card-catalog-modal__footer'>
+            <label class='card-catalog-modal__url-search'>
+              <span>Enter Card URL:</span>
+              <BoxelInput @value={{this.cardURL}} placeholder='http://' />
+            </label>
+            <div>
+              <Button
+                @kind='secondary-light'
+                @size='tall'
+                @disabled={{eq this.selectedCard undefined}}
```
Comment (lukemelia) [inline packages/host/app/components/card-catalog-modal.gts]: It seems like the Cancel button should still be available even if no card has been selected.

**Review by lukemelia — COMMENTED** [review]
Perhaps doubleclick on a card should pick it, without having to press Go?

**Pull request merged**